### DISCUSSION TITLE
Cooker domain - adding support

### DISCUSSION THREAD

**Discussion by matchett808-gh** [home-assistant/architecture#945] [Entity Models]
Hi,

I've got an air fryer that is Tuya based, local Tuya doesn't work for this as there isn't really a mapping to the native functionality - I'd like to add a domain and then an integration that does the following: 

- Finish time (stored as finish time but set as run for x minutes) 
- temperature
- presets
- power
- keep warm
- preheat 

Presets follow a similar idea to modes for vacuums, keep warm and preheat kindof map to switches (but only after a run is 'pushed' to the air fryer) and there is no good map for duration and temperature

My initial thought is to add an air fryer or cooker component - but I sort of feel that ApplianceWithRuntimeAndPresets might be a better domain (my dishwasher would fit this description and I expect many large and medium kitchen appliances would) - so possibly 'Appliance' as a domain that has support for the following:

- Run (based on settings, singular click/action) (required)
- pause (optional, not all devices will support this)
- presets/modes (required - fries, eco50, cupboard dry etc - up to the integration whether this is set on the device at this point or later when run is pressed)
- additional optional features (keep warm, preheat, extra dry for my dishwasher - possibly freeform? but optional and up to the integration whether this is set on the device at this point or later when run is pressed) 
- temperature (optional - up to the integration whether this is set on the device at this point or later when run is pressed)
- runtime in minutes (optional - single or bi-directiona and up to the integration whether this is set on the device at this point or later when run is pressedl)

This would likely lead to the domain exposing the following services:
- appliance.run
- appliance.pause?
- appliance.setMode: <preset>
- appliance.setOption: <option: value>
   - appliance.setOption: <temperature: value>
   - appliance.setOption: <runTime: value>

Alternatively it'd be really neat if there's not any appetite within core to have these domains for custom components to be able to supply these as required

This may solve other requests - it may also open up new use cases as devices not currently easily supportable could be massaged into this slightly more freeform implementation.

**Comment by matchett808-gh** [comment]
To add - I'm happy to start working on this, but discussion first ofc

**Comment by gjohansson-ST** [comment]
An suggestion for a `cooker` entity component already exist in #501 so please continue the discussion there.
I will close this one.
